src/Extensions/Compatability/Validation/NotNullWhenAttribute.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#if NETFRAMEWORK || NETSTANDARD2_0
```

Comment:
I really hope there is no need in using it for single class, it will add more confusion. And considering that using exclude condition is not a common approach I would prefer not to use it in open source if possible.